From 4dd9e8fd585210a31bd8a99630a4d958d794ab28 Mon Sep 17 00:00:00 2001
From: Chris Mayo <aklhfex@gmail.com>
Date: Sat, 18 May 2024 17:13:31 +0100
Subject: Fix bashism in configure tests for curl and libcmis

When /bin/sh is dash:

./configure: 35165: test: yes: unexpected operator
checking whether to enable breakpad... no
./configure: 35218: test: yes: unexpected operator

Causing HAVE_FEATURE_CURL not to be set to 1 and build to fail:

sw/source/ui/misc/translatelangselect.cxx:160:24: error: no member named 'TranslateDocumentCancellable' in namespace 'SwTranslateHelper'
  160 |     SwTranslateHelper::TranslateDocumentCancellable(m_rWrtSh, aConfig, m_bCancelTranslation);
      |     ~~~~~~~~~~~~~~~~~~~^

Introduced in:

5bf7c2fa5794 ("Fix --disable-curl build", 2023-09-14)

Change-Id: Ifbc4bc1a7cde86101ff13b05ec7cee6836798605
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/167831
Tested-by: Ilmari Lauhakangas <ilmari.lauhakangas@libreoffice.org>
Reviewed-by: Ilmari Lauhakangas <ilmari.lauhakangas@libreoffice.org>
Tested-by: Jenkins
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index fbd6196921a5..59f1c97dbef7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -11079,7 +11079,7 @@ AC_SUBST(ICU_MINOR)
 dnl ==================================================================
 dnl CURL
 dnl ==================================================================
-if test "$enable_curl" == "yes"; then
+if test "$enable_curl" = "yes"; then
     AC_DEFINE([HAVE_FEATURE_CURL])
 fi
 
@@ -11125,7 +11125,7 @@ AC_SUBST(DEFAULT_CRASHDUMP_VALUE)
 dnl ==================================================================
 dnl libcmis
 dnl ==================================================================
-if test "$enable_libcmis" == "yes"; then
+if test "$enable_libcmis" = "yes"; then
     if test "$enable_curl" != "yes"; then
         AC_MSG_ERROR([--disable-libcmis must be used when --disable-curl is used])
     fi
-- 
cgit v1.2.3

